src/contents/en-gb/doc_xamarin.md

```diff
@@ -0,0 +1,210 @@
+```csharp
+public const string AppId = "net.maidsafe.examples.todo";
+public const string AppName = "Safe Todo";
+var authReq = new AuthReq
+{
+    AppContainer = true,
+    App = new AppExchangeInfo 
+    { 
+        Id = AppId, 
+        Scope = "", 
+        Name = AppName, 
+        Vendor = "MaidSafe.net Ltd." 
+    },
+    Containers = new List<ContainerPermissions> 
+    { 
+        new ContainerPermissions 
+        { 
+            ContName = "_publicNames", Access = { Insert = true, Update = true, Delete = true } 
+        }
+    }
+};
+var (_, reqMsg) = await Session.EncodeAuthReqAsync(authReq);
+var ipcReq = await authenticator.DecodeIpcMessageAsync(reqMsg);
+var authIpcReq = ipcReq as AuthIpcReq;
+var resMsg = await authenticator.EncodeAuthRespAsync(authIpcReq, true);
+var ipcResponse = await Session.DecodeIpcMessageAsync(resMsg);
+var authResponse = ipcResponse as AuthIpcMsg;
+var session = await Session.AppRegisteredAsync(appId, authResponse.AuthGranted);
+```
+
+## Connecting to Test SAFE Network
+
+To be able to connect to the SAFE Network, a SAFE application needs to get an authorisation from the user, this is achieved by sending an authorisation request to the Authenticator (Safe Authenticator app in this case).
+
+We first need to generate an `AuthReq` instance which provides information about the application and, permissions requested by this application. These details will be displayed in the authenticator:
+
+```csharp
+public const string AppId = "net.maidsafe.examples.todo";
+public const string AppName = "Safe Todo";
+var authReq = new AuthReq
+{
+    AppContainer = true,
+    App = new AppExchangeInfo 
+    { 
+        Id = AppId, 
+        Scope = "", 
+        Name = AppName, 
+        Vendor = "MaidSafe.net Ltd." 
+    },
+    Containers = new List<ContainerPermissions> 
+    { 
+        new ContainerPermissions 
+        { 
+            ContName = "_publicNames", Access = { Insert = true, Update = true, Delete = true } 
+        }
+    }
+};
+```
+Note that we are passing a list of `ContainerPermissions`, this allows us to define the various permissions provided to specific containers.
+
+Once the `AuthReq` instance is initialised, we can use it to generate the authorisation request and send it to the Authenticator using following code:
+```csharp
+(uint, string) encodedReq = await Session.EncodeAuthReqAsync(authReq);
+var url = $"safe-auth://{AppId}/{encodedReq}";
+Device.BeginInvokeOnMainThread(() => { Device.OpenUri(new Uri(url)); });
+```
+Make sure you have Authenticator App installed and logged in before running the application.
+
+Once app request is sent, you would notice that Safe Authenticator app is launched and a pop-up dialogue is prompting for access. At this point, you can allow or deny the request, and the Authenticator will send the response back to the application. 

```

Comment:
Once `the` app request is sent, you ~would~ `will` notice ~that~ `the` Safe Authenticator app is launched and a pop-up dialogue is prompting for access. 
wording changes ^^